### ISSUE TITLE
Implementation of Marching Tetrahedra

### ISSUE THREAD

**Issue by KhoiDOO** [KhoiDOO/conquer3d#1]
- CUDA-based implementation.
- Tetrahedras are obtained from splitting voxels.
- Convention: XZY, where Z is vertical-axis.

**Comment by KhoiDOO** [comment]
Reference: https://paulbourke.net/geometry/polygonise/

**Comment by KhoiDOO** [comment]
### Usage Example: CUDA Marching Tetrahedra with Differentiability

```python
import torch
from conquer3d.ops import marching_tetrahedra, tetrahedralize

device = torch.device("cuda:0")

# 1. Prepare points & tetrahedral mesh
pts = torch.randn((10000, 3), device=device)
tets = tetrahedralize(pts) # or construct from regular/sparse voxel grid

# 2. Define scalar SDF field (requires_grad for differentiable optimization)
sdfs = (torch.norm(pts, dim=-1) - 0.5).requires_grad_(True)
colors = torch.rand((pts.shape[0], 3), device=device, requires_grad=True)

# 3. Extract surface using CUDA Marching Tetrahedra
verts, faces, verts_colors = marching_tetrahedra(pts, tets, sdfs, colors=colors, iso=0.0)

# 4. Compute loss and backpropagate directly through CUDA kernel
loss = (verts ** 2).sum() + (verts_colors ** 2).sum()
loss.backward()

print("SDF Gradient:", sdfs.grad.shape)
print("Extracted Mesh:", verts.shape, faces.shape)
```

**Comment by KhoiDOO** [comment]
Implemented and verified via CUDA Marching Tetrahedra pipeline in PR #7.
